File: src/main/java/org/pl/interpreter/visitor/NumberVisitor.java
package org.pl.interpreter.visitor;

import org.pl.interpreter.IInterpreter;
import org.pl.interpreter.exception.VisitorException;
import org.pl.parser.ast.INode;
import org.pl.parser.ast.NumberNode;

import java.math.BigDecimal;


/**
 * Interprets a number node.
 */
public class NumberVisitor implements IVisitor {

    @Override
    public boolean matches(INode node) {
        return NumberNode.class == node.getClass();
    }

    @Override
    public BigDecimal visit(IInterpreter interpreter, INode node) throws VisitorException {
        return ((NumberNode) node).value;
    }
}

File: src/main/kotlin/org/leaflang/interpreter/visitor/NumberVisitor.kt
package org.leaflang.interpreter.visitor

import org.leaflang.interpreter.IInterpreter
import org.leaflang.interpreter.result.numberResult
import org.leaflang.parser.ast.INode
import org.leaflang.parser.ast.NumberNode

/**
 * Interprets a number node.
 */
class NumberVisitor : IVisitor {
    override fun visit(interpreter: IInterpreter, node: INode) = numberResult((node as NumberNode).value)
}